pwiz_tools/Shared/Common/SystemUtil/ProcessRunner.cs

```diff
@@ -53,23 +53,48 @@ public class ProcessRunner : IProcessRunner
         public string HideLinePrefix { get; set; }
 
         public void Run(ProcessStartInfo psi, string stdin, IProgressMonitor progress, ref IProgressStatus status,
-            ProcessPriorityClass priorityClass = ProcessPriorityClass.Normal)
+            ProcessPriorityClass priorityClass = ProcessPriorityClass.Normal, bool forceTempfilesCleanup = false)
         {
-            Run(psi, stdin, progress,ref status, null, priorityClass);
+            Run(psi, stdin, progress,ref status, null, priorityClass, forceTempfilesCleanup);
         }
 
         public void Run(ProcessStartInfo psi, string stdin, IProgressMonitor progress, ref IProgressStatus status, TextWriter writer,
-            ProcessPriorityClass priorityClass = ProcessPriorityClass.Normal)
+            ProcessPriorityClass priorityClass = ProcessPriorityClass.Normal, bool forceTempfilesCleanup = false)
         {
             // Make sure required streams are redirected.
             psi.RedirectStandardOutput = true;
             psi.RedirectStandardError = true;
             psi.RedirectStandardInput = stdin != null;
             if (OutputEncoding != null)
                 psi.StandardOutputEncoding = psi.StandardErrorEncoding = OutputEncoding;
+            string tmpDirForCleanup = null;
 
             _messageLog.Clear();
 
+            if (forceTempfilesCleanup)

```

Comment:
This appears to be another place where a sub-function might improve readability rather than adding a new chunk of code to an already large function dedicated to a single issue.

Reply:
No problem.